File: tropo_whitelist.py
from subprocess import check_output


def print_whitelist():
    '''Retrieves tropo's whitelist and prints to STDOUT.
    '''
    response = get_tropo_whitelist_response()
    if response:
        lines = response.decode().split("\n")
        for line in lines:
            if "\"" in line:
                csv_line = format_cidr_to_csv(line)
                print(csv_line)


def format_cidr_to_csv(cidr):
    cidr_block = cidr.split("\"")[1]
    return "tropo," + cidr_block


def get_tropo_whitelist_response():
    '''Retrieves Tropo's whitelist using "nslookup" tool through CLI

    Returns: string
    '''
    return check_output(["nslookup", "-q=TXT",
                         "_netblocks.tropo.com", "8.8.8.8"])

if __name__ == '__main__':
    print_whitelist()

File: tropo_whitelist.rb
# Retrieves Tropo's current IP info for whitelisting
class TropoWhitelist
  def self.print_whitelist
    response = self.get_tropo_whitelist_response
    if response
      response.split("\n").each do |line|
        cidr_block = line.split('"')[1]
        puts "tropo," + cidr_block if cidr_block
      end
    end
  end

  private

  # Requires that nslookup tool be installed on host computer
  def self.get_tropo_whitelist_response
    `nslookup -q=TXT _netblocks.tropo.com 8.8.8.8`
  end
end

TropoWhitelist.print_whitelist
